ros/node.go

```diff
@@ -41,21 +40,13 @@ type defaultNode struct {
 	waitGroup      sync.WaitGroup
 }
 
-func listenRandomPort(address string, trialLimit int) (net.Listener, error) {
-	var listener net.Listener
-	var err error
-	numTrial := 0
-	for numTrial < trialLimit {
-		port := 1024 + rand.Intn(65535-1024)
-		addr := fmt.Sprintf("%s:%d", address, port)
-		listener, err = net.Listen("tcp", addr)
-		if err == nil {
-			return listener, nil
-		} else {
-			numTrial += 1
-		}
+func listenRandomPort(address string) (net.Listener, error) {
+	addr := fmt.Sprintf("%s:0", address)
+	if listener, err := net.Listen("tcp", addr); err == nil {
+		return listener, nil
+	} else {
+		return nil, fmt.Errorf("listenRandomPort cannot found a free port.")
```

Comment:
Maybe it would be better to print "err" in a log message, rather then just printing a not so informative string.